src/mlpack/core/tree/cover_tree/cover_tree_impl.hpp

```diff
@@ -876,6 +893,23 @@ CoverTree<MetricType, StatisticType, MatType, RootPointPolicy>::
       furthestDescendantDistance;
 }
 
+
+template<
+    typename MetricType,
+    typename StatisticType,
+    typename MatType,
+    typename RootPointPolicy
+>
+typename CoverTree<MetricType, StatisticType, MatType,
+    RootPointPolicy>::ElemType
+CoverTree<MetricType, StatisticType, MatType, RootPointPolicy>::
+    MaxDistance(const arma::vec& other, const ElemType bestDistance, bool forOverload)
```

Comment:
I think the signature is wrong here, should this be `MaxDistanceForFast`?  (same with the other `MaxDistance` overload)